craft_store/base_client.py

```diff
@@ -186,10 +186,23 @@ def upload_file(
         """Upload filepath to storage.
 
         The monitor_callback is a method receiving one argument of type
-        ``MultipartEncoderMonitor``, among others, it's ``.bytes_read`` and
-        ``.len`` attributes can be used to track progress.
+        ``MultipartEncoder``, this method is to return a method to track
+        progress. Among others, it's ``.bytes_read`` and ``.len`` attributes
```

Comment:
This sounds a bit confusing, should that "it's" be "its"? And what object is supposed to have these attributes?